sklearn/ensemble/_hist_gradient_boosting/splitting.pyx

```diff
@@ -504,12 +528,13 @@ cdef class Splitter:
                             feature_idx, histograms, n_samples,
                             sum_gradients, sum_hessians,
                             value, monotonic_cst[feature_idx],
-                            lower_bound, upper_bound, &split_infos[feature_idx])
+                            lower_bound, upper_bound, &split_infos[split_info_idx])
 
             # then compute best possible split among all features
-            best_feature_idx = self._find_best_feature_to_split_helper(
-                split_infos)
-            split_info = split_infos[best_feature_idx]
+            # split_info is the index of the best split_info
```

Comment:
I think this should be 

```suggestion
            # split_info is the best split_info
```

as it's not an index, but I'm not sure this comment is needed?